hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ClusteringPlanActionExecutor.java

```diff
@@ -77,11 +78,14 @@ protected Option<HoodieClusteringPlan> createClusteringPlan() {
       return Option.empty();
     }
 
-    LOG.info("Generating clustering plan for table " + config.getBasePath());
-    ClusteringPlanStrategy strategy = (ClusteringPlanStrategy)
-        ReflectionUtils.loadClass(ClusteringPlanStrategy.checkAndGetClusteringPlanStrategy(config), table, context, config);
+    ClusteringPlanStrategy strategy = null;
+    if (config.getAsyncClusterMaxCommits() <= commitsSinceLastClustering) {
+      LOG.info("Generating clustering plan for table " + config.getBasePath());
+      strategy = (ClusteringPlanStrategy)
+          ReflectionUtils.loadClass(ClusteringPlanStrategy.checkAndGetClusteringPlanStrategy(config), table, context, config);
+    }
 
-    return strategy.generateClusteringPlan();
+    return strategy == null ? Option.empty() : strategy.generateClusteringPlan();
```

Comment:
@eric9204 This does not seem to solve the problem you mentioned, around frequent clustering scheduling.  This only avoids NPE.